### ISSUE TITLE
非方形缩放问题

### ISSUE THREAD

**Issue by lazyn1997** [PaddlePaddle/PaddleClas#1524]
我的训练数据长宽比在2-3之间，所以我想使用非方形缩放，将 ResizeImage.size设置为[800, 320]，会报如下图的错误
![image](https://user-images.githubusercontent.com/34644177/144694232-2b27f28a-8817-42f9-8fcc-e286aa7f2a9d.png)
请问这个错误应该如何解决？

**Comment by RainFrost1** [comment]
能否提供完整的yaml文件，我们测试一下

**Comment by lazyn1997** [comment]
`# global configs
Global:
  checkpoints: null
  pretrained_model: null
  output_dir: "./output/LYC_filter_280+251/"
  device: "gpu"
  class_num: 531
  save_interval: 1
  eval_during_train: True
  eval_interval: 1
  epochs: 200
  print_batch_step: 10
  use_visualdl: True
  # used for static mode and model export
  image_shape: [3, 800, 320]
  save_inference_dir: "./inference/LYC_filter_280+251/"
  eval_mode: "retrieval"

# model architecture
Arch:
  name: "RecModel"
  infer_output_key: "features"
  infer_add_softmax: False
  Backbone: 
    name: "ResNet50_last_stage_stride1"
    pretrained: True
  BackboneStopLayer:
    name: "adaptive_avg_pool2d_0"
  Neck:
    name: "VehicleNeck"
    in_channels: 2048
    out_channels: 512
  Head:
    name: "ArcMargin"  
    embedding_size: 512
    class_num: 531
    margin: 0.15
    scale: 32
 
# loss function config for traing/eval process
Loss:
  Train:
    - CELoss:
        weight: 1.0
    - SupConLoss:
        weight: 1.0
        views: 2
    - TripletLossV2:
        weight: 1.0
        margin: 0.5
  Eval:
    - CELoss:
        weight: 1.0

Optimizer:
  name: Momentum
  momentum: 0.9
  lr:
    name: Cosine
    learning_rate: 0.02
    warmup_epoch: 5
  regularizer:
    name: 'L2'
    coeff: 0.0005

# data loader for train and eval
DataLoader:
  Train:
    dataset:
        name: "VeriWild"
        image_root: "G:/postgraduate_data/feature/"
        cls_label_path: "G:/postgraduate_data/feature/LYC_filter_train_list_280+251.txt"
        transform_ops:
          - DecodeImage:
              to_rgb: True
              channel_first: False
          - ResizeImage:
              size: [800, 320]
          - RandFlipImage:
              flip_code: 1
          - AugMix:
              prob: 0.5
          - NormalizeImage:
              scale: 0.00392157
              mean: [0.485, 0.456, 0.406]
              std: [0.229, 0.224, 0.225]
              order: ''
          - RandomErasing:
              EPSILON: 0.5
              sl: 0.02
              sh: 0.4
              r1: 0.3
              mean: [0., 0., 0.]

    sampler:
        name: PKSampler
        batch_size: 64
        sample_per_id: 2
        drop_last: False
        shuffle: True
    loader:
        num_workers: 0
        use_shared_memory: True
  Eval:
    Query:
      dataset: 
        name: "VeriWild"
        image_root: "G:/postgraduate_data/feature/"
        cls_label_path: "G:/postgraduate_data/feature/LYC_filter_test_list_280+251.txt"
        transform_ops:
          - DecodeImage:
              to_rgb: True
              channel_first: False
          - ResizeImage:
              size: [800, 320]
          - NormalizeImage:
              scale: 0.00392157
              mean: [0.485, 0.456, 0.406]
              std: [0.229, 0.224, 0.225]
              order: ''
      sampler:
        name: DistributedBatchSampler
        batch_size: 64
        drop_last: False
        shuffle: False
      loader:
        num_workers: 0
        use_shared_memory: True

    Gallery:
      dataset: 
        name: "VeriWild"
        image_root: "G:/postgraduate_data/feature/"
        cls_label_path: "G:/postgraduate_data/feature/LYC_filter_test_list_280+251.txt"
        transform_ops:
          - DecodeImage:
              to_rgb: True
              channel_first: False
          - ResizeImage:
              size: [800, 320]
          - NormalizeImage:
              scale: 0.00392157
              mean: [0.485, 0.456, 0.406]
              std: [0.229, 0.224, 0.225]
              order: ''
      sampler:
        name: DistributedBatchSampler
        batch_size: 64
        drop_last: False
        shuffle: False
      loader:
        num_workers: 0
        use_shared_memory: True

Metric:
  Eval:
    - Recallk:
        topk: [1, 5]
    - mAP: {}

`
大概就是上面这样，只是把size的值由224改为了[800, 320]，就会报上述的错误，我想的就是把图片缩放为长方形进行训练，应该如何操作？

**Comment by lazyn1997** [comment]
```
# global configs
Global:
checkpoints: null
pretrained_model: null
output_dir: "./output/LYC_filter_280+251/"
device: "gpu"
class_num: 531
save_interval: 1
eval_during_train: True
eval_interval: 1
epochs: 200
print_batch_step: 10
use_visualdl: True

used for static mode and model export
image_shape: [3, 800, 320]
save_inference_dir: "./inference/LYC_filter_280+251/"
eval_mode: "retrieval"

model architecture
Arch:
name: "RecModel"
infer_output_key: "features"
infer_add_softmax: False
Backbone:
name: "ResNet50_last_stage_stride1"
pretrained: True
BackboneStopLayer:
name: "adaptive_avg_pool2d_0"
Neck:
name: "VehicleNeck"
in_channels: 2048
out_channels: 512
Head:
name: "ArcMargin"
embedding_size: 512
class_num: 531
margin: 0.15
scale: 32

loss function config for traing/eval process
Loss:
Train:
- CELoss:
weight: 1.0
- SupConLoss:
weight: 1.0
views: 2
- TripletLossV2:
weight: 1.0
margin: 0.5
Eval:
- CELoss:
weight: 1.0

Optimizer:
name: Momentum
momentum: 0.9
lr:
name: Cosine
learning_rate: 0.02
warmup_epoch: 5
regularizer:
name: 'L2'
coeff: 0.0005

data loader for train and eval
DataLoader:
Train:
dataset:
name: "VeriWild"
image_root: "G:/postgraduate_data/feature/"
cls_label_path: "G:/postgraduate_data/feature/LYC_filter_train_list_280+251.txt"
transform_ops:
- DecodeImage:
to_rgb: True
channel_first: False
- ResizeImage:
size: [800, 320]
- RandFlipImage:
flip_code: 1
- AugMix:
prob: 0.5
- NormalizeImage:
scale: 0.00392157
mean: [0.485, 0.456, 0.406]
std: [0.229, 0.224, 0.225]
order: ''
- RandomErasing:
EPSILON: 0.5
sl: 0.02
sh: 0.4
r1: 0.3
mean: [0., 0., 0.]

sampler:
    name: PKSampler
    batch_size: 64
    sample_per_id: 2
    drop_last: False
    shuffle: True
loader:
    num_workers: 0
    use_shared_memory: True
Eval:
Query:
dataset:
name: "VeriWild"
image_root: "G:/postgraduate_data/feature/"
cls_label_path: "G:/postgraduate_data/feature/LYC_filter_test_list_280+251.txt"
transform_ops:
- DecodeImage:
to_rgb: True
channel_first: False
- ResizeImage:
size: [800, 320]
- NormalizeImage:
scale: 0.00392157
mean: [0.485, 0.456, 0.406]
std: [0.229, 0.224, 0.225]
order: ''
sampler:
name: DistributedBatchSampler
batch_size: 64
drop_last: False
shuffle: False
loader:
num_workers: 0
use_shared_memory: True

Gallery:
  dataset: 
    name: "VeriWild"
    image_root: "G:/postgraduate_data/feature/"
    cls_label_path: "G:/postgraduate_data/feature/LYC_filter_test_list_280+251.txt"
    transform_ops:
      - DecodeImage:
          to_rgb: True
          channel_first: False
      - ResizeImage:
          size: [800, 320]
      - NormalizeImage:
          scale: 0.00392157
          mean: [0.485, 0.456, 0.406]
          std: [0.229, 0.224, 0.225]
          order: ''
  sampler:
    name: DistributedBatchSampler
    batch_size: 64
    drop_last: False
    shuffle: False
  loader:
    num_workers: 0
    use_shared_memory: True
Metric:
Eval:
- Recallk:
topk: [1, 5]
- mAP: {}
```
大概就是上面这样，只是把size的值由224改为了[800, 320]，就会报上述的错误，我想的就是把图片缩放为长方形进行训练，应该如何操作？

**Comment by RainFrost1** [comment]
好的，我验证一下

**Comment by RainFrost1** [comment]
已经定位到问题，是图像预处理中AugMix的bug，具体修复可以查看pr#[1536](https://github.com/PaddlePaddle/PaddleClas/pull/1536)

**Comment by lazyn1997** [comment]
感谢回复，已经修改，可以跑通了
